clang/lib/Sema/SemaDecl.cpp

```diff
@@ -10540,8 +10540,20 @@ Sema::ActOnFunctionDeclarator(Scope *S, Declarator &D, DeclContext *DC,
         HasExplicitTemplateArgs = false;
       } else if (FunctionTemplate) {
         // Function template with explicit template arguments.
+#ifdef SYCLomatic_CUSTOMIZATION
+
+        if (getLangOpts().CUDA) {
```

Comment:
Please check if need add this condition.

Reply:
`nvcc -x c++` cannot compile this code also. So it is reasonable to add this condition.